samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -149,10 +151,13 @@ public void runTask() {
 
   @Override
   public void run(StreamApplication app) {
+    SystemAdmins systemAdmins = null;
     try {
-      super.run(app);
+      systemAdmins = buildAndStartSystemAdmins();
```

Comment:
seems this three lines are in quite a few places. I suggest hiding the systemAdmins inside the streamManager, add add start(), stop() in streamManager so we can invoke life cycle methods of systemAdmins. With this, we can still make streamManager a member in the abstract runner to avoid constructing it everywhere.

Reply:
Updated to put SystemAdmins inside StreamManager, and then manage the StreamManager lifecycle inside the runners.